glue/viewers/image/composite_array.py

```diff
@@ -137,7 +138,8 @@ def __call__(self, bounds=None):
             data = interval(array)
             data = contrast_bias(data, out=data)
             data = stretch(data, out=data)
-            data[np.isnan(data)] = 0
+            if self._set_nans_to_zero:
+                data[np.isnan(data)] = 0
```

Comment:
This was only introduced in fa60bf1 for compatibility with Astropy; if we can reproduce what the issue with Astropy was, preserving NaN may perhaps be made the default again.

Reply:
@astrofrog Any memory/tips of what we should check? Thanks!

Reply:
I *think* it was actually that astropy did not preserve NaNs and then started to so it's not an issue with astropy as such - but more that when you do color compositing you don't really have the option to give NaN values their own color since each channel has only one color assigned to it. So I think it's as I mentioned on the call - for multi-color compositing we should always set NaNs to 0 and for the colormap we could never set it to zero.